drivers/SmartThings/matter-switch/src/inovelli-vtm31-sn/init.lua

```diff
@@ -0,0 +1,301 @@
+}
+
+local function is_inovelli_vtm31_sn(opts, driver, device)
+  if device.network_type == device_lib.NETWORK_TYPE_MATTER and
+     device.manufacturer_info.vendor_id == INOVELLI_VTM31_SN_FINGERPRINT.vendor_id and
+     device.manufacturer_info.product_id == INOVELLI_VTM31_SN_FINGERPRINT.product_id then
+    log.info("Using Inovelli VTM31 sub driver")
+    return true
+  end
+  return false
+end
+
+local preferences_to_numeric_value = function(new_value)
+  local numeric = tonumber(new_value)
+  if numeric == nil then -- in case the value is Boolean
+    numeric = new_value and 1 or 0
+  end
+  return numeric
+end
+
+local preferences_calculate_parameter = function(new_value, type, number)
+  if number == "parameter9" or number == "parameter10" or number == "parameter15" then
+    if new_value == 101 then
+      return 255
+    else
+      return utils.round(new_value / 100 * 254)
+    end
+  end
+  return new_value
+end
+
+local function to_boolean(value)
+  if value == 0 or value == "0" then
+    return false
+  end
+  return true
+end
+
+local function get_first_non_zero_endpoint(endpoints)
+  for _,ep in ipairs(endpoints) do
+    if ep ~= 0 then -- 0 is the matter RootNode endpoint
+      return ep
+    end
+  end
+  return nil
+end
+
+local function find_default_endpoint(device)
+  local switch_eps = device:get_endpoints(clusters.OnOff.ID)
+  table.sort(switch_eps)
+  local main_endpoint = get_first_non_zero_endpoint(switch_eps)
+  if main_endpoint ~= nil then
+    return main_endpoint
+  end
+  device.log.warn(string.format("Did not find default endpoint, will use endpoint %d instead", device.MATTER_DEFAULT_ENDPOINT))
+  return device.MATTER_DEFAULT_ENDPOINT
+end
+
+local function find_child(parent, ep_id)
+  return parent:get_child_by_parent_assigned_key(string.format("%d", ep_id))
+end
+
+local function initialize_buttons_and_switches(driver, device, main_endpoint)
+  local switch_eps = device:get_endpoints(clusters.OnOff.ID)
+  local button_eps = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH})
+  local component_map = {}
+  local current_component_number = 1
+  -- Since we do not support bindings at the moment, we only want to count clusters
+  -- that have been implemented as server. This can be removed when we have
+  -- support for bindings.
+  local num_switch_server_eps = 0
+  device:try_update_metadata({profile = "inovelli-vtm31-sn"})
+  -- The first switch endpoint will be the main component and the three buttons
+  -- will be added as additional components in a MCD profile.
+  component_map["main"] = main_endpoint
+  table.sort(button_eps)
+  for _, ep in ipairs(button_eps) do
+    component_map[string.format("button%d", current_component_number)] = ep
+    current_component_number = current_component_number + 1
+  end
+  device:set_field(COMPONENT_TO_ENDPOINT_MAP, component_map, {persist = true})
+  configure_buttons.configure_buttons(device)
```

Comment:
can we rename things such that we end with:
```suggestion
  button_capability.configure(device)
```
Where we perhaps rename the .lua file itself? This could be better for extending the generic button functionality into there as well, which I think could be a good idea in general.

Reply:
or maybe `buttons` instead of `button_capability`. At first I thought button_capability might be clearer, but I'm not too worried about that, and it's a bit of a mouthful.

Reply:
Sure thing - I changed it to `buttons` in the latest commit